documentation/src/main/asciidoc/topics/con_cache_encoding.adoc

```diff
@@ -0,0 +1,41 @@
+[id='encoding']
+= Cache encoding
+Encoding is the format, or media type, that {brandname} uses to store entries (key/value pairs) in caches.
+
+[discrete]
+== Remote caches
+
+{brandname} Server stores entries in remote caches without any media type information by default.
+
+Encoding configures how {brandname} stores entries in a remote cache.
+If you do not configure encoding for remote caches then {brandname} Server stores entries however clients provide them.
+In this case {brandname} Server also returns entries as they are stored in the remote cache for any client requests.
+With encoding, {brandname} converts keys and values, when necessary, to whatever media type you specify for the cache.
+
+[TIP]
+====
+Always configure an encoding for remote caches if you want to use multiple clients, such as {brandname} Console or CLI, Hot Rod, REST, as well as server-side tasks and scripts.
+
+{brandname} recommends application/x-protostream`.
+====
+
+[discrete]
+== Embedded caches
+
+{brandname} stores entries in embedded caches as Plain Old Java Objects (POJOs) by default.
+
+Storing POJOs in embedded caches means that object instances are mutable.
+For example if you store a `HashMap` any modifications to the map are seen by other users of the cache, causing race conditions.
```

Comment:
I'm not sure what the mutability of POJOs stored in local caches has todo with marshalling? I think we can remove this paragraph and start the next paragraph as "For clustered embedded ... "

Reply:
This is because the POJO can be accessed and changed as a get(...) will return a reference to the POJO.
If now the cache is clustered and ISPN decides to marshall it for replication it cause rase conditions (like ConncurrentModification for Collection, or inconsistent replication)
So I think it is important to document this as user will run into this

Reply:
Ok I understand now. I think this is an important thing to explain, however I think we need to explain this cenario in more detail and consequently I don't think it should be one of the first things we mention. I think we should first explain that serialization is necessary (currently 3rd paragraph), then have a note explaining that mutable POJOs stored in the cache must be updated by value using a new POJO instance, in order to avoid race conditions when performing serialization.

Reply:
@ryanemerson and @wfink could you please check the updated wording addresses the review comments?

Reply:
> For example if you store a `HashMap` any modifications to the map are seen by other users of the cache, which leads to race conditions when serialization is performed.

This still isn't correct. How about:

"For example, if you store a `HashMap` as a key/value, any local modifications to the Map are not seen by other cluster members. Furthermore, it's possible that a ConcurrentModificationException could occur if the Map instance is updated whilst the object is being marshalled by {brandname}.

Reply:
thanks @ryanemerson I've updated the wording here.